server/odc-service/src/main/java/com/oceanbase/odc/service/session/ConnectConsoleService.java

```diff
@@ -341,6 +435,32 @@ public List<SqlExecuteResult> getAsyncResult(@NotNull String sessionId, String r
         }
     }
 
+    public AsyncExecuteResultResp getAsyncResultV2(@NotNull String sessionId, String requestId) {
+        PreConditions.validArgumentState(Objects.nonNull(requestId), ErrorCodes.SqlRegulationRuleBlocked, null, null);
+        ConnectionSession connectionSession = sessionService.nullSafeGet(sessionId);
+        AsyncExecuteContext context =
+                (AsyncExecuteContext) ConnectionSessionUtil.getExecuteContext(connectionSession, requestId);
+        List<JdbcGeneralResult> resultList = context.getResults();
+        List<SqlExecuteResult> results = resultList.stream().map(jdbcGeneralResult -> {
+            SqlExecuteResult result = generateResult(connectionSession, jdbcGeneralResult, context.getContextMap());
+            try (TraceStage stage = result.getSqlTuple().getSqlWatch().start(SqlExecuteStages.SQL_AFTER_CHECK)) {
+                sqlInterceptService.afterCompletion(result, connectionSession, context.getContextMap());
+            } catch (Exception e) {
+                throw new IllegalStateException(e);
+            }
+            return result;
+        }).collect(Collectors.toList());
+        if (context.isFinished()) {
```

Comment:
add an try-catch-finally here and put 

```java
if (context.isFinished) {
    ConnectionSessionUtil.removeExecuteContext(connectionSession, requestId);
}
```

in finally block to make sure that this statement will be executed in ant case~